src/Thentos/DB/Trans.hs

```diff
@@ -244,38 +264,36 @@ trans_resetPassword now token newPass = do
                             let user' = userPassword .~ newPass $ user
                             _ <- writeUser uid user'
                             returnDb label ()
-                        Nothing ->
-                            throwDb label NoSuchResetToken
+                        Nothing -> throwDb label NoSuchToken
 
 resetTokenExpiryPeriod :: Timeout
 resetTokenExpiryPeriod = Timeout 3600
 
--- | Update existing user in DB.  Throw an error if user id does not
--- exist, or if email address in updated user is already in use by
--- another user.
-trans_updateUser :: UserId -> User -> ThentosUpdate ()
-trans_updateUser uid user = do
-    ThentosLabeled label  _  <- liftThentosQuery $ trans_lookupUser uid
-    ThentosLabeled label' () <- writeUser uid user
-    returnDb (lub label label') ()
-
 data UpdateUserFieldOp =
     UpdateUserFieldName UserName
   | UpdateUserFieldEmail UserEmail
   | UpdateUserFieldAddService ServiceId
   | UpdateUserFieldDropService ServiceId
-  deriving (Eq, Show)
+  | UpdateUserFieldPassword (HashedSecret UserPass)
+  deriving (Eq)
 
+-- | Update existing user in DB.  Throw an error if user id does not
+-- exist, or if email address in updated user is already in use by
+-- another user.
 trans_updateUserField :: UserId -> UpdateUserFieldOp -> ThentosUpdate ()
-trans_updateUserField uid op = do
+trans_updateUserField uid op = trans_updateUserFields uid [op]
+
+trans_updateUserFields :: UserId -> [UpdateUserFieldOp] -> ThentosUpdate ()
+trans_updateUserFields uid ops = do
```

Comment:
nice idea.  this way we can still choose to change any number of fields inside one transaction, but we don't have to.  this makes `updateUser` completely unnecessary.